### PR TITLE
Add utility for single-node parallel installations

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR adds a utility for doing single-node parallel installations using backgrounded instances of `spack install`. Mostly suitable for small installations or interactive batch jobs; for larger installations using a batch job along the lines of what we have for Acorn probably makes more sense.

### Testing

Tested on personal machine.

### Applications affected

none

### Systems affected

none

### Dependencies

none

### Issue(s) addressed

none

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Review by climbfuji — COMMENTED** [review]
What is the return code from this script? Is it /= 0 if one or more of the parallel install commands fail?

**Comment by AlexanderRichert-NOAA** [comment]
Based on the `wait` command, the return code is the most recent one, so we probably want something more robust than that. I'll play it around with it some time. This definitely doesn't need to be there for 1.6.0, just a nice to have.

**Review by srherbener — APPROVED** [review]
Thanks for adding this utility. It looks quite useful!

File: util/parallel_install.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+argtext="Run as '$0 N M', where N=# of instances, M=# of threads per instance"
+n_instances=${1?"$argtext"}
+n_threads=${2?"$argtext"}
+shift 2
+
+if [[ ! " $(seq -s ' ' 10) " =~ " $n_instances " ]]; then echo "Invalid number of instances" ; exit 1; fi
+if [[ ! " $(seq -s ' ' 64) " =~ " $n_threads " ]]; then echo "Invalid number of threads" ; exit 1; fi
+
+echo "Installing with $n_instances instances and ${n_threads} threads in environment '${SPACK_ENV?"SPACK_ENV not set!"}'"
```
Comment (climbfuji) [inline util/parallel_install.sh]: Do you want to abort if SPACK_ENV is not set, i.e. make it a requirement?

**Comment by climbfuji** [comment]
Also add to doc/source/?

File: util/parallel_install.sh (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply util/parallel_install.sh]: I think that's what this will do? If not then I can explicitly set `set -a` or whatever to fail on non-zero.

Comment (climbfuji) [reply util/parallel_install.sh]: `set -e` I think - I was confused because the notes above say "it is strongly recommended" to do this in an active env.

**Review by climbfuji — APPROVED** [review]
Thanks, looks good to me!

**Review by ulmononian — APPROVED** [review]

**Comment by climbfuji** [comment]
I resolved the conflict, once the readthedocs page is rendered I'll take a look, if it looks good I'll merge

**Pull request merged**